pep-0677.rst

```diff
@@ -826,6 +826,27 @@ replacement for callable types:
   more like a shorter way to write them than a replacement for
   ``Callable``.
 
+Hybrid keyword-arrow Syntax
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the Rust language, a keyword ``fn`` is used to indicate functions
+in much the same way as Python's ``def``, and callable types are
+indicated using a hybrid arrow syntax ``Fn(i64, String) -> bool``.
+
+We could use the ``def`` keyword in callable types for Python, for
+example our two-parameter boolean function could be written as
+``def(int, str) -> bool``. But we think this might confuse readers
+into thinking ``def(A, B) -> C`` is a lambda, particularly because
+Javascript's ``function`` keyword is used in both named and anonymous
+functions.
+
+We could alternatively use ``Def(int, str) -> bool``, where the
+capital letter strongly suggests a type rather than a lambda. But
+that would require a new keyword, and we do not see any benefits that
+justify adding a new keyword or making the types more verbose than
+the current proposal.
```

Comment:
I would drop this paragraph (on the 'Def' keyword), unless someone other than you actually brought that up on a public forum. It's just not something anyone would propose. (but using 'def' totally would be).